src/operation/insert/mod.rs

```diff
@@ -96,20 +110,61 @@ impl<T: Serialize> Operation for Insert<T> {
             .into());
         }
 
-        let mut body = doc! {
-            Self::NAME: self.ns.coll.clone(),
-            "documents": docs,
+        let mut options = self.options.clone().unwrap_or_default();
+        options.ordered = Some(self.is_ordered());
+
+        let body = InsertCommand {
+            insert: self.ns.coll.clone(),
+            documents: DocumentArraySpec {
+                documents: docs,
+                length: size as i32,
+            },
+            options,
         };
 
-        append_options(&mut body, self.options.as_ref())?;
+        Ok(Command::new("insert".to_string(), self.ns.db.clone(), body))
+    }
+
+    fn serialize_command(&mut self, cmd: Command<Self::Command>) -> Result<Vec<u8>> {
```

Comment:
Since we already have raw BSON for the documents, we can't just use serde to serialize it, as it'll attempt to do so as a a `Binary`. Instead, we serialize the rest of the command and manually append the serialized documents. Again, this will be easier with the raw document API.